src/data/logbook.ts

```diff
@@ -72,6 +106,166 @@ const getLogbookDataFromServer = async (
   return hass.callApi<LogbookEntry[]>("GET", url);
 };
 
-export const clearLogbookCache = (startDate, endDate) => {
+export const clearLogbookCache = (startDate: string, endDate: string) => {
   DATA_CACHE[`${startDate}${endDate}`] = {};
 };
+
+export const getLogbookMessage = (
+  hass: HomeAssistant,
+  state: string,
+  stateObj: HassEntity,
+  domain: string
+): string => {
+  const localizePath = "ui.components.logbook.messages";
+
+  switch (domain) {
+    case "device_tracker":
+    case "person":
+      return state === "not_home"
+        ? hass.localize(`${localizePath}.was_away`)
+        : hass.localize(`${localizePath}.was_at_state`, "state", state);
+
+    case "sun":
+      return state === "above_horizon"
+        ? hass.localize(`${localizePath}.rose`)
+        : hass.localize(`${localizePath}.set`);
+
+    case "binary_sensor": {
+      const isOn = state === BINARY_STATE_ON;
+      const isOff = state === BINARY_STATE_OFF;
+      const device_class = stateObj.attributes.device_class;
+
+      switch (device_class) {
+        case "battery":
+          if (isOn) {
+            return hass.localize(`${localizePath}.was_low`);
+          }
+          if (isOff) {
+            return hass.localize(`${localizePath}.was_normal`);
+          }
+          break;
+
+        case "connectivity":
+          if (isOn) {
+            return hass.localize(`${localizePath}.was_connected`);
+          }
+          if (isOff) {
+            return hass.localize(`${localizePath}.was_disconnected`);
+          }
+          break;
+
+        case "door":
+        case "garage_door":
+        case "opening":
+        case "window":
+          if (isOn) {
+            return hass.localize(`${localizePath}.was_opened`);
```

Comment:
Can we make the localize names:
`ui.components.logbook.messages.${domain}.${device_class || "_"}.${state}`

That's what we also use for the state translations, and requires a lot less code.

Reply:
The state translations are built by core for all the states. Is it even possible to do that here? like in the code there are a lot of if this state do this otherwise this. Doing it your way would require me to do this for all states? no?

Reply:
You would move the logic to the translations instead of the code, not more complicated I guess?

Reply:
How do I do the logic in Json?